### PR TITLE
Reproducible images using a sequence of seeds

### PR DESCRIPTION
**Pull request by JelleAalbers**
This PR makes it possible to reproduce datasets, and individual images from training datasets (without generating all previous images). This should be useful e.g. to generate converge maps, or the list of subhalo masses and positions, for specific images in datasets we are interested in.

Paltas already  allows you to set the numpy random seed via the config. However, it also draws some random numbers from numba, which uses a separate random number generator. Thus, setting the seed in paltas' config does not yet result in predictable images. This also means some of the unit tests are not fully deterministic.

After this PR: 
  * We re-seed the numpy random number generator before generating each image, and seed the separate numba random generator right after that.
  * The first generated image uses a base seed. The second uses a tuple (base_seed, 1) as a seed (or base_seed + (1,) if base_seed is already a tuple), then (base_seed, 2), etc. Numpy uses [SeedSequence](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.SeedSequence.html#numpy.random.SeedSequence) internally to turn these low-quality (not very random) seeds into high-quality seeds, see e.g. [here](https://numpy.org/doc/stable/reference/random/bit_generators/mt19937.html#mersenne-twister-mt19937).
  * If the user doesn't specify a base seed to use in the config, `ConfigHandler` will set the random number generator seed to a random int64. (We could use a larger seed, but this way seeds are still easy to copy-paste from metadata.)

We're not exactly following best practice with random number generators: current wisdom avoids `np.seed` and instead makes separate random generator instances and passes them around the code. That would be a much bigger change affecting the entire code though.

Two minor changes smuggled in:
  * When creating the folder with npy images files, allow the folder to already exist. The mkdir can be triggered if the folder already exists, when the catalog's main npy is missing (for some reason).
  * Do not create a tfrecord during the generate tests, so we can run all the non-analysis tests in environments without tensorflow. The tfrecord was not actually tested anyway (other than implicity requiring it exists by the os.remove cleanup statement). If you'd rather the file is made during testing we could e.g. check if tf imports, and only ask for the tfrecord in the tests if it does.

### CODE REVIEW

File: paltas/Configs/config_handler.py [hunk 1]
```diff
@@ -36,17 +37,25 @@ class ConfigHandler():
 		config_path (str): A path to the config file to parse.
 	"""
 
-	def __init__(self,config_path):
+	reseed_counter = 0
```
Comment (swagnercarena) [inline paltas/Configs/config_handler.py:40]: Having this be a property of the class is a bit odd here. Would there be any time when we would want to access reseed_counter without initializing an instance of the class?

File: test/generate_tests.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def test_main(self):
 		n_generate = 21
 		output_folder = 'test_data/test_dataset'
 		sys.argv = ['test','test_data/config_dict.py',output_folder,'--n',
-			str(n_generate),'--tf_record','--save_png_too']
```
Comment (swagnercarena) [inline test/generate_tests.py:36]: I understand the logic here, but we still want to include something that runs the tf_record in the analysis tests then. Otherwise we'll lose coverage from never checking those lines.

File: paltas/Configs/config_handler.py (hunk 1 above)
Comment (JelleAalbers) [reply paltas/Configs/config_handler.py]: OK, setting it explicitly as an instance variable in __init_ now

File: test/generate_tests.py (hunk 2 above)
Comment (JelleAalbers) [reply test/generate_tests.py:36]: OK, now added back --tf_record to the argv if tensorflow imports

**Review by swagnercarena — APPROVED** [review]

**Comment by swagnercarena** [comment]
Looks great!
